frontend/src/app/hooks/error-boundary.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {
+  Component,
+  useState,
+  useCallback,
+  createContext,
+  useContext,
+  MutableRefObject,
+  useMemo,
+  useRef,
+  ComponentType,
+  ReactNode,
+  PropsWithChildren,
+  ReactElement,
+  ErrorInfo,
+} from "react";
+
+type ComponentDidCatch = (error: Error, errorInfo: ErrorInfo) => void;
+
+interface ErrorBoundaryProps {
+  error: Error | undefined;
+  onError: ComponentDidCatch;
+}
+
+class ErrorBoundary extends Component<PropsWithChildren<ErrorBoundaryProps>> {
+  displayName = "UseErrorBoundary";
+
+  componentDidCatch(...args: Parameters<NonNullable<Component["componentDidCatch"]>>) {
+    // silence React warning:
+    // ErrorBoundary: Error boundaries should implement getDerivedStateFromError().
+    // In that method, return a state update to display an error message or fallback UI
+    this.setState({});
+    this.props.onError(...args);
+  }
+
+  render() {
+    return this.props.children;
+  }
+}
+
+const noop = () => false;
+
+interface ErrorBoundaryCtx {
+  componentDidCatch: MutableRefObject<ComponentDidCatch | undefined>;
+  error: Error | undefined;
+  errorInfo: ErrorInfo | undefined;
+  setError: (error: Error | undefined) => void;
+}
+
+const errorBoundaryContext = createContext<ErrorBoundaryCtx>({
+  componentDidCatch: { current: undefined },
+  error: undefined,
+  errorInfo: undefined,
+  setError: noop,
+});
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export function ErrorBoundaryContext({ children }: { children?: ReactNode }) {
+  const [error, setError] = useState<Error>();
```

Comment:
Any reason to not do what sonarcloud is recommending?  Looks like this is a legit best practice for React  

```suggestion
export function ErrorBoundaryContext({ children }: { readonly children?: ReactNode | undefined }) {
```

Reply:
yeah there's no reason for sonar cloud to flag this as props are readonly by default.Sonar cloud has been updated to close this error out